packages/ERTP/src/mathHelpers/natListMathHelpers.js

```diff
@@ -0,0 +1,43 @@
+import harden from '@agoric/harden';
+import { passStyleOf } from '@agoric/marshal';
+import { assert } from '@agoric/assert';
+import Nat from '@agoric/nat';
+
+// Operations for arrays with unique Nat elements.
+const identity = harden([]);
+
+const natListMathHelpers = harden({
+  doAssertKind: list => {
+    assert(passStyleOf(list) === 'copyArray', 'list must be an array');
+    list.forEach(Nat);
+  },
+  doGetIdentity: _ => identity,
+  doIsIdentity: list => passStyleOf(list) === 'copyArray' && list.length === 0,
+  doIsGTE: (left, right) => {
+    const leftSet = new Set(left);
+    const has = elem => leftSet.has(elem);
+    return right.every(has);
+  },
+  doIsEqual: (left, right) => {
+    left = [...left];
+    right = [...right];
+    left.sort();
+    right.sort();
+    return left.every((num, i) => num === right[i]);
+  },
+  doAdd: (left, right) => {
```

Comment:
In light of https://github.com/Agoric/agoric-sdk/issues/557 we should at least instrument this to see when the union is not disjoint, and try to understand why that happens. Once we understand this better, we might want to make this error if left and right intersect. Doesn't need to happen in this PR, but if not, please file an issue or extend #557

Reply:
I'll file an issue specifically for this.